components/tidb_query/src/rpn_expr/impl_op.rs

```diff
@@ -452,4 +461,25 @@ mod tests {
             assert_eq!(output, expected);
         }
     }
+
+    #[test]
+    fn test_right_shift() {
+        let cases = vec![
+            (Some(123), Some(2), Some(30)),
+            (Some(-123), Some(-1), Some(0)),
+            (Some(123), Some(0), Some(123)),
+            (None, Some(1), None),
+            (Some(123), None, None),
+            (Some(-123), Some(2), Some(4611686018427387873)),
+            (None, None, None),
```

Comment:
Add a case that will cause an `"overflow"`?

Reply:
Why? It doesn't make sense...

Reply:
It means we should add a test case that checks this `shift` operation is a `wrapping` shift or a `checked` shift